pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -763,19 +763,19 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 				og.recorder.Eventf(pvcWithResizeRequest.PVC, v1.EventTypeWarning, kevents.VolumeResizeFailed, updateErr.Error())
 				return updateErr
 			}
-		}
-
-		// No Cloudprovider resize needed, lets mark resizing as done
-		if !volumePlugin.RequiresFSResize() {
-			glog.V(4).Infof("Controller resizing done for PVC %s", pvcWithResizeRequest.QualifiedName())
-			err := resizeMap.MarkAsResized(pvcWithResizeRequest, newSize)
 
-			if err != nil {
-				glog.Errorf("Error marking pvc %s as resized : %v", pvcWithResizeRequest.QualifiedName(), err)
-				og.recorder.Eventf(pvcWithResizeRequest.PVC, v1.EventTypeWarning, kevents.VolumeResizeFailed, err.Error())
-				return err
+			// No Cloudprovider resize needed, lets mark resizing as done
+			if !volumePlugin.RequiresFSResize() {
+				glog.V(4).Infof("Controller resizing done for PVC %s", pvcWithResizeRequest.QualifiedName())
+				err := resizeMap.MarkAsResized(pvcWithResizeRequest, newSize)
+				if err != nil {
+					glog.Errorf("Error marking pvc %s as resized : %v", pvcWithResizeRequest.QualifiedName(), err)
+					og.recorder.Eventf(pvcWithResizeRequest.PVC, v1.EventTypeWarning, kevents.VolumeResizeFailed, err.Error())
+					return err
+				}
 			}
 		}
+
```

Comment:
log a message here with both newSize and pv size and this resize should not be executed.

Reply:
Done, thanks